core/src/main/java/com/chopshop166/chopshoplib/logging/LoggedSubsystem.java

```diff
@@ -0,0 +1,59 @@
+package com.chopshop166.chopshoplib.logging;
+
+import org.littletonrobotics.junction.Logger;
+import org.littletonrobotics.junction.inputs.LoggableInputs;
+import com.chopshop166.chopshoplib.commands.SmartSubsystemBase;
+
+/**
+ * A subsystem that supports logging via inheritance.
+ * 
+ * @param <D> The data object for input/output.
+ * @param <M> The hardware map.
+ */
+public abstract class LoggedSubsystem<D extends LoggableInputs, M extends LoggableMap<D>>
+        extends SmartSubsystemBase {
+
+    /** The input/output data. */
+    private final D ioData;
+    /** The hardware map. */
+    private final M map;
+
+    /**
+     * Constructor.
+     * 
+     * @param ioData The data object to use for i/o.
+     * @param map The hardware mapping.
+     */
+    public LoggedSubsystem(final D ioData, final M map) {
+        super();
+        this.ioData = ioData;
```

Comment:
Can we have another constructor that creates ioData internally? I don't expect the Data class to have an actual constructor most of the time.

Reply:
We cannot - thanks Java generics